### ISSUE TITLE
scrubber: more robust metadata consistency check

### ISSUE THREAD

**Issue by jcsp** [neondatabase/neon#8128]
The scrubber's scan_metadata command will flag some inconsistencies, but it's not quite robust enough to trust at scale:
- It needs to handle objects being deleted while it scans, by re-reading the index after scan and checking that any objects that vanished have also vanished from the index.
- It needs a way of reporting results that we can alert on at scale.  Perhaps this is a log, perhaps it's a write to a database, or a write to a scrub report in S3.

Basically the code in `branch_cleanup_and_check_errors` is somewhat old and crufty and needs a fresh set of eyes to make sure we are robustly checking a well understood set of things, before we can use it at scale to increase confidence that our metadata's integrity is okay.

```[tasklist]
### Tasks
- [ ] #8344
- [ ] https://github.com/neondatabase/neon/pull/8424
- [ ] https://github.com/neondatabase/neon/issues/8421
- [ ] https://github.com/neondatabase/neon/pull/8480
- [ ] https://github.com/neondatabase/neon/pull/8502
- [ ] https://github.com/neondatabase/neon/pull/8565
- [ ] https://github.com/neondatabase/neon/pull/8635
- [x] Wire the check into a command suitable for running along with physical GC (a la https://github.com/neondatabase/cloud/issues/14024)
- [ ] https://github.com/neondatabase/neon/issues/8872
- [ ] Setup alert on metadata health status
- [ ] scrubber: invoke storage controller to learn about timeline existence (blocked until storage controller is timeline-aware)
```

**Comment by yliang412** [comment]
Last week:
- write out race conditions between RemoteTimelineClient and scrubber.

This week:
- implement changes to `scan_metadata` to make the checks robust.

**Comment by yliang412** [comment]
Last week:
- implemented the check (without considering parallelism to head object calls).

This week:
- add some helper functions for creating corrupted S3 state + check if scrubber actually detect errors
- add it to the same cronjob as Physical GC (or maybe a seperate cronjob?)

**Comment by yliang412** [comment]
Last week: 
- Opened  #8480 
- Opened #8502 

This week:
- Merge existing PRs
- Wire `scan_metadata` command into scrubber cronjob.

**Comment by yliang412** [comment]
Last week: 
- Opened #8565 

This week:
- Wire scan_metadata command into scrubber cronjob.

**Comment by yliang412** [comment]
Last week:
- Merged #8635 

This week: 
- Update helm-charts to use unified command.

**Comment by yliang412** [comment]
Last week:

- Found a new transient error as a result of race between scrubber listing and timeline deletion [[Grafana](https://neonprod.grafana.net/d/8jHyEQOna/neon-logs-storage-scrubber?orgId=1&var-environment=dev&var-datasource=xHHYY0dVz&var-search=&var-exclude=substing%20to%20exclude&var-region=eu-west-1)].
  - Classify errors and see if any are non-fatal
  - We will switch to and exit-code based alerting (@skyzh set up the alert [here](https://neonprod.grafana.net/alerting/grafana/fdsy1x5pw17ggb/view)).

This week:
- Merge #8873

**Comment by skyzh** [comment]
> We will switch to and exit-code based alerting

already done, let's refine the exit code within scrubber to make the alert more informative

**Comment by yliang412** [comment]
Last week:
- Handle concurrent `list_timeline_blobs` and timeline deletion: https://github.com/neondatabase/neon/pull/8873

This week:
- Look into scrubber logs to see if there are any failures.
